tokio-console/src/state/tasks.rs

```diff
@@ -217,16 +218,38 @@ impl TasksState {
                     warnings: Vec::new(),
                     location,
                 };
-                task.lint(linters);
+                let recheck = task.lint(linters);
+                if recheck {
+                    self.pending_lint.insert(task.id);
+                }
                 Some((id, task))
             });
 
+        let mut checked = HashSet::new();
         for (stats, mut task) in self.tasks.updated(stats_update) {
             tracing::trace!(?task, ?stats, "processing stats update for");
             task.stats = stats.into();
-            task.lint(linters);
+            let recheck = task.lint(linters);
+            if !recheck {
+                self.pending_lint.remove(&task.id);
+            }
+            checked.insert(task.id);
```

Comment:
Switch the logic here (part of **PendingLintLogic**, see below).
```suggestion
            if recheck {
                next_pending_lint.insert(task.id);
            }
            self.pending_lint.remove(&task.id);
```